WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -250,11 +250,11 @@ public ICoinsView AsAllCoinsView()
 	{
 		lock (Lock)
 		{
-			return new CoinsView(AsAllCoinsViewNoLock().ToList());
```

Comment:
It doesn't seem good.

Returning an IEnumerable inside a lock without materializing it (e.g., by calling ToList()) can be risky because:

Deferred Execution: IEnumerable executes lazily, meaning the actual data retrieval happens when you iterate over it. If the underlying data changes outside the lock after you've returned the IEnumerable, you may encounter unexpected behavior or exceptions when you later iterate over it.

Thread Safety: If the underlying collection is modified by another thread while being iterated over, it can lead to concurrency issues. By not materializing the collection inside the lock, you lose control over when the data is accessed, potentially leading to thread-safety problems.

In summary, returning an IEnumerable inside a lock without materializing it can lead to deferred execution and thread-safety issues, making the code more prone to errors and harder to maintain.